clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/HttpElasticsearchDAO.scala

```diff
@@ -54,6 +58,33 @@ class HttpElasticsearchDAO private[dataaccess] (
     searchResponse flatMap foldScroll(Seq.empty[D], searchDefinition.keepAlive)
   }
 
+  /**
+    * Given an elastic search index, return the most recent document for that index.
+    *
+    * Document ordering is determined by the "upsert ID", which must be indexed field.
+    *
+    * @param index the elasticsearch index
+    * @tparam D the document type this index contains
+    * @return the most recent document for this index, if any
+    */
+  def getMostRecentDocument[D: HitReader](
+    index: ElasticsearchIndex[D]
+  ): Future[D] = {
+    val searchDefinition = search(index.indexName / index.indexType)
+      .size(1)
+      .sortByFieldDesc(ClioDocument.UpsertIdElasticSearchName)
+    for {
+      searchResponse <- httpClient execute searchDefinition
+    } yield
+      searchResponse.to[D] match {
```

Comment:
Nitpick: can you throw another set of braces around this whole `yield` body? They're redundant and IntelliJ complains about them but in general I think it really helps readability to wrap any multi-line statement in them.